consensus/src/step_votes_reg.rs

```diff
@@ -177,18 +190,18 @@ impl CertInfoRegistry {
 
     pub(crate) fn get_nil_certificates(
         &mut self,
-        from: usize,
-        to: usize,
+        to: u8,
     ) -> Vec<Option<Certificate>> {
-        let to = std::cmp::min(to, self.cert_list.len());
-        let mut res = Vec::with_capacity(to - from);
-
-        for item in &self.cert_list[from..to] {
-            if item.is_nil() {
-                res.push(Some(item.cert));
-            } else {
-                res.push(None)
-            }
+        let mut res = Vec::with_capacity(to as usize);
+
+        for iteration in 0u8..to {
+            res.push(
+                self.cert_list
+                    .get(&iteration)
+                    .map(|c| c.nil)
+                    .filter(|ci| ci.is_nil_quorum())
```

Comment:
Are you excluding partial (non-quorum) certificates from FailedIterations?

Reply:
Yes, I didn't change it.

Ps: you were right then